openvdb/openvdb/points/PointRasterizeSDF.h

```diff
@@ -0,0 +1,1701 @@
+        for (auto leafIter = range.begin(); leafIter; ++leafIter) {
+            const int min = this->minDist(minacc.getValue(leafIter->origin()));
+            if (min <= 0) continue;
+            this->fill(*leafIter, min, false);
+        }
+    }
+
+private:
+    inline int maxDist(const typename RadiusTreeT::ValueType& max) const
+    {
+        // max radius in index space
+        const Real v = (Real(max) * mScale) + mHalfband;
+        int d = int(math::RoundUp(v)); // next voxel
+        d += this->mVoxelOffset; // final offset
+        return d;
+    }
+
+    inline int minDist(const typename RadiusTreeT::ValueType& min) const
+    {
+        // min radius in index space
+        Real v = math::Max(0.0, (Real(min) * mScale) - mHalfband);
+        v = math::Max(0.0, Real(v - this->mVoxelOffset)); // final offset
+        // calculate the min interior cube area of activity. this is the side
+        // of the largest possible cube that fits into the radius "v":
+        //   d = 2r -> 2r = 3x^2 -> x = 2r / sqrt(3)
+        // finally half it to the new radius
+        v = ((2.0 * v) / std::sqrt(3.0)) / 2.0;
+        return static_cast<int>(v); // round down (v is always >= 0)
+    }
+
+private:
+    const RadiusTreeT& mMin;
+    const RadiusTreeT& mMax;
+    const Real mScale, mHalfband;
+};
+
+template <typename SdfT, typename InterrupterT, typename PointDataGridT>
+inline typename SdfT::Ptr
+initFixedSdf(const PointDataGridT& points,
+        math::Transform::Ptr transform,
+        const typename SdfT::ValueType bg,
+        const double minBandRadius,
+        const double maxBandRadius,
+        InterrupterT* interrupter)
+{
+    using LeafManagerT = tree::LeafManager<const typename PointDataGridT::TreeType>;
+    using MaskTreeT = typename SdfT::TreeType::template ValueConverter<ValueMask>::Type;
+
+    typename SdfT::Ptr surface = SdfT::create(bg);
+    surface->setTransform(transform);
+    surface->setGridClass(GRID_LEVEL_SET);
+
+    FixedSurfaceMaskOp<MaskTreeT, InterrupterT> op(points.transform(),
+       *transform, minBandRadius, maxBandRadius, interrupter);
+
+    if (interrupter) interrupter->start("Generating uniform surface topology");
+
+    LeafManagerT leafManager(points.tree());
+    tbb::parallel_reduce(leafManager.leafRange(), op);
+    auto mask = op.mask();
+
+    if (minBandRadius > 0.0) {
+        auto maskoff = op.maskoff();
+        mask->topologyDifference(*maskoff);
+        // union will copy empty nodes so prune them
+        tools::pruneInactive(*mask);
+        surface->tree().topologyUnion(*mask);
+        mask.reset();
+        // set maskoff values to -background
+        tree::ValueAccessor<const MaskTreeT> acc(*maskoff);
+        auto setOffOp = [acc](auto& iter) {
+            if (acc.isValueOn(iter.getCoord())) {
+                iter.modifyValue([](auto& v) { v = -v; });
+            }
+        };
+        tools::foreach(surface->beginValueOff(), setOffOp,
```

Comment:
Unless I'm mis-reading this, isn't this using the slow tree value iterator?

Reply:
Yes this should really use the dynamic node manager but it didn't exist when I wrote this